### PR TITLE
feat: custom actions for user maintenance and debug triggers

### PR DESCRIPTION
**Pull request by phorward**
This provides user maintenance triggers to either kick users from the system or login as any user. This feature is only available for "root" users.

The PR enables to use translate()-objects in adminInfos as well.

### CODE REVIEW

File: core/modules/user.py [hunk 1]
```diff
@@ -942,6 +988,30 @@ def getAuthMethods(self, *args, **kwargs):
...
+        match action:
+            case "takeover":
+                self.authenticateUser(skel["key"])
+
+            case "kick":
+                session.killSessionByUser(skel["key"])
+
+            case _:
+                raise errors.NotAcceptable(f"Action {action!r} not implemented")
+
```
Comment (sveneberth) [inline core/modules/user.py:1012]:
```suggestion
current.request.get().response.headers["Content-Type"] = "application/json"
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by ArneGudermann — APPROVED** [review]

File: core/modules/user.py [hunk 2]
```diff
@@ -942,6 +988,30 @@ def getAuthMethods(self, *args, **kwargs):
 
         return json.dumps(res)
 
+    @exposed
+    def trigger(self, action, key, skey):
```
[the remark this answers is not in the document]
Comment (sveneberth) [reply core/modules/user.py]: @ArneGudermann This cannot work. If you add a type hint to a exposed method it must be unambiguous to be type-cast-able by `processTypeHint`.

This results now in 
```py
ValueError: TypeHint str | viur.datastore.types.Key not supported
```

File: core/modules/user.py [hunk 3]
```diff
@@ -946,6 +992,30 @@ def getAuthMethods(self, *args, **kwargs):
 
         return json.dumps(res)
 
+    @exposed
+    def trigger(self, action: str, key: str | db.Key, skey: str):
```
Comment (sveneberth) [inline core/modules/user.py:996]:
```suggestion
    def trigger(self, action: str, key: str, skey: str):
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Comment by phorward** [comment]
Just as hint, the adminInfo can be subclasses this way now to inherit the actions:

```python
class User(User):

    adminInfo = User.adminInfo | {
        "columns": ["lastname", "firstname", "name", "responsibility"],
        "filter": {"orderby": "lastname"},
        "disabledActions": [
            "add",
            "clone",
            "delete"
        ],
    }
```

File: core/modules/user.py (hunk 1 above)
Comment (sveneberth) [reply core/modules/user.py:1019]: @phorward Hmm, This isn't resolved?!

File: core/modules/user.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArneGudermann) [reply core/modules/user.py]: You are right. But the  `processTypeHint` function should still support `|` in a Typehint.

[the remark this answers is not in the document]
Comment (ArneGudermann) [reply core/modules/user.py]: Maybe the `processTypeHint` function should provide `|` in the Typehint.

[the remark this answers is not in the document]
Comment (sveneberth) [reply core/modules/user.py]: Let's discuss this in this issue: https://github.com/viur-framework/viur-core/issues/718

File: core/modules/user.py [hunk 4]
```diff
@@ -946,6 +992,33 @@ def getAuthMethods(self, *args, **kwargs):
 
         return json.dumps(res)
 
+    @exposed
+    def trigger(self, action: str, key: str, skey: str):
+        current.request.get().response.headers["Content-Type"] = "application/json"
+
+        # only "root" users are allowed to trigger maintenance functions!
+        if not ((cuser := current.user.get()) and "root" in cuser["access"]):
```
Comment (sveneberth) [inline core/modules/user.py:1000]: This should be configurable in the adminInfo.

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: core/modules/user.py (hunk 4 above)
Comment (phorward) [reply core/modules/user.py:1000]: But how?

Comment (sveneberth) [reply core/modules/user.py:1000]: I'm not sure about the meaning of access within custom access. Does one of the access roles have to be satisfied or all of them?


But something like this should do it?
```suggestion
        access = self.adminInfo[action]["access"]
        if not ((cuser := current.user.get()) and any(role in cuser["access"] for in access):
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Comment by phorward** [comment]
@sveneberth I've implemented your recommendations and merged, updated and tested against current develop.
Can we now merge?

**Review by sveneberth — APPROVED** [review]
With the last vi version it works smoothly.

**Pull request merged**